### PR TITLE
add activation ops under paddle.nn and paddle.nn.functional: ReLU, LogSoftmax

### PR DESCRIPTION
**Pull request by zhupengyang**
- add 4 api
    - paddle.nn.ReLU(inplace=None)
    - paddle.nn.functional.relu(input, inplace=False, name=None)
    - paddle.nn.LogSoftmax(axis=None)
    - paddle.nn.functional.log_softmax(input, axis=None, dtype=None, name=None)
- relu的inplace说明
    - relu的反向过程只需要out和dout
    - inplace在动态图不支持，只输出warning，参考reshape对inplace的处理
- LogSoftmax
    - 不在paddle.nn.LogSoftmax开放dtype参数，参考Softmax

### CODE REVIEW

File: python/paddle/nn/functional/activation.py [hunk 1]
```diff
@@ -12,29 +12,170 @@
...
+
+    if in_dygraph_mode():
+        if inplace:
+            warnings.warn(
+                "Inplace on ReLU is not allowed and will be discarded in dygraph mode currently."
+            )
+        outs = core.ops.relu(inputs)
+        return outs['Out'][0]
+
+    helper = LayerHelper("fc", **locals())
```
Comment (hong19860320) [inline python/paddle/nn/functional/activation.py]: 为什么这里使用fc？

Comment (zhupengyang) [reply python/paddle/nn/functional/activation.py]: Done。写错了。。

File: python/paddle/nn/functional/activation.py [hunk 2]
```diff
@@ -12,29 +12,167 @@
...
+    helper.append_op(
+        type='softmax',
+        inputs={'X': outs_cast},
+        outputs={'Out': outs_softmax},
+        attrs={'axis': axis,
+               'use_cudnn': False})
+
+    outs_log = helper.create_variable_for_type_inference(outs_softmax.dtype)
+    helper.append_op(
+        type='log', inputs={'X': outs_softmax}, outputs={'Out': outs_log})
```
Comment (qingqing01) [inline python/paddle/nn/functional/activation.py:176]: LogSoftmax is maily for numerical stability, why not implement a C++ op?
